File: two_pointers/can_place_flowers.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        zeroes = 0
        for f in [0] + flowerbed + [0]:
            if f == 0:
                zeroes += 1
            else:
                zeroes = 0

            if zeroes == 3:
                zeroes = 1
                n -= 1

            if n == 0:
                break

        return n == 0

File: two_pointers/can_place_flowers.rb
# @param {Integer[]} flowerbed
# @param {Integer} n
# @return {Boolean}
def can_place_flowers(flowerbed, n)

    zeroes = 0
    ([0] + flowerbed + [0]).each do |spot|

        if spot == 0
            zeroes += 1
        else
            zeroes = 0
        end

        if zeroes == 3
            zeroes = 1
            n -= 1
        end

        break if n == 0
    end
    
    n == 0
end